lib/matplotlib/axes/_axes.py

```diff
@@ -5578,13 +5578,13 @@ def _interp_grid(X):
                         dX = np.diff(X, axis=1)/2.
                         if not (np.all(dX >= 0) or np.all(dX <= 0)):
                             cbook._warn_external(
-                                f"The input coordinates to {funcname} are "
-                                "not all either increasing or decreasing, "
-                                "the automatically computed edges can "
+                                f"Cell edges are being inferred from mesh "
+                                f"cell centers, but the input coordinates to "
+                                f"{funcname} are not monotonically increasing "
+                                "or decreasing. The inferred cell edges can "
                                 "produce misleading results in this case. "
-                                "It is recommended to supply the "
-                                f"quadrilateral edges to {funcname}"
-                                f" yourself. See help({funcname}).")
+                                "If so, it is possible to supply the "
```

Comment:
I don't know what the condition is for this 'if', but I think it's:
```suggestion
                                "If is is possible, supply the "
```

Reply:
If the results are misleading...

Reply:
reworded a bit..